src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/ExceptionHandling.cs

```diff
@@ -771,7 +771,13 @@ private static void DispatchEx(scoped ref StackFrameIterator frameIter, ref ExIn
 
                 DebugScanCallFrame(exInfo._passNumber, frameIter.ControlPC, frameIter.SP);
 
-                UpdateStackTrace(exceptionObj, exInfo._frameIter.FramePointer, (IntPtr)frameIter.OriginalControlPC, frameIter.SP, ref isFirstRethrowFrame, ref prevFramePtr, ref isFirstFrame, ref exInfo);
+#if !NATIVEAOT
```

Comment:
Why do we want NATIVEAOT behavior to be different?

Reply:
I am actually not sure if the issue occurs with NativeAOT too. It uses a different way of filtering frames in the UpdateStackTrace based on equal frame pointers, which doesn't work on coreclr.  So I've made the change for coreclr, but plan to investigate the nativeaot case.

Reply:
What does the diagnostic test that this fixing does?

Reply:
It is the NestedException2 test. Basically, there is this code:
```c#
        public static void Foo()
        {
            try
            {
                Console.WriteLine("Foo calling Bar");
                ExceptionsTestHelper.Bar<int>(1);
            }
            catch (Exception inner)
            {
                throw new Exception("doh", inner);
            }
        }
    }
```
The `ExceptionsTestHelper.Bar<int>(1);` just throws `InvalidOperationException`. The issue is that the `Exception` thrown in the code above has the line of `ExceptionsTestHelper.Bar<int>(1);` on the stack trace.

Reply:
> The issue is that the Exception thrown in the code above has the line of ExceptionsTestHelper.Bar<int>(1); on the stack trace.

I see the ` My.Main()` line duplicated:
```
C:\runtime\artifacts\bin\coreclr\windows.x64.Release>corerun test.exe
Foo calling Bar
Unhandled exception. System.Exception: doh
 ---> System.InvalidOperationException: Operation is not valid due to the current state of the object.
   at My.Bar[T](Int32 x)
   at My.Main()
   --- End of inner exception stack trace ---
   at My.Main()

C:\runtime\artifacts\bin\coreclr\windows.x64.Release>set DOTNET_LegacyExceptionHandling=0

C:\runtime\artifacts\bin\coreclr\windows.x64.Release>corerun test.exe
Foo calling Bar
Unhandled exception. System.Exception: doh
 ---> System.InvalidOperationException: Operation is not valid due to the current state of the object.
   at My.Bar[T](Int32 x)
   at My.Main()
   --- End of inner exception stack trace ---
   at My.Main()
   at My.Main()
```

There is ifdefed-out code in UpdateStackTrace that seems to be dealing with this situation:

https://github.com/dotnet/runtime/blob/68f7885f19b56bdc0045b87aae944c66ee9f0694/src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/ExceptionHandling.cs#L966-L968

Can this bug be fixed by deleting the ifdef in `UpdateStackTrace`?

Reply:
No, that's what I've mentioned above. The frame pointer check doesn't work in coreclr, since there are cases where two non-funclet methods have the same frame pointer - when they don't use Rbp as a frame pointer and use just Rsp instead. I've actually attempted to fix the problem by removing the ifdef, thinking I was originally wrong. But then another diagnostic test started to fail due the fact I've just mentioned.